### ISSUE TITLE
UP032 became less useful in v0.3.1

### ISSUE THREAD

**Issue by ThiefMaster** [astral-sh/ruff#10259]
The whole change in that rule to trigger even when it causes a longer line feels kind of annoying and the fact that there's no autofix doesn't make it better: Fixing it requires manual work anyway, but I'd rather start from a too-long line which I just need to wrap somehow, than having to do the whole fix myself. --> Couldn't it simply be an unsafe autofix in such cases?

Some example where just using f-string isn't great / very readable:

```py
return '{} - {}'.format(
    format_skeleton(start_dt, day_month, timezone=tzinfo),
    format_skeleton(end_dt, day_month, timezone=tzinfo)
)
```

(of course this one can easily be refactored to separate statements and then a nicely compact f-string)

```py
        db.CheckConstraint('(input_type IS NULL) = (type NOT IN ({t.field}, {t.field_pd}))'
                           .format(t=RegistrationFormItemType),
```

In this case an f-string do not provide the same level of functionality. And this happens to be part of an sqlalchemy `__table_args__` class attribute, so moving the assignment to a separate statement would clutter the class scope (or require an explicit `del t` afterwards).

**Comment by mmerickel** [comment]
Ran into this too with a bunch of tedious fixes. An unsafe autofix would be great to just have to deal with the line length afterward. Running the `pyupgrade` tool directly doesn't trigger on these lines either so ruff is being more strict here which is fine but an autofix would just be really helpful.

**Comment by charliermarsh** [comment]
I've enabled the autofix for these new violations here: https://github.com/astral-sh/ruff/pull/10263.

**Comment by charliermarsh** [comment]
Part of the problem is we don't really have a mechanism for gating by line length -- it's not really the intent of fix safety, which is intended to capture changes in the semantics and meaning of the code. So in that PR, I've made it a safe fix regardless. There's some discussion on the broader issue here, if anyone has opinions: https://github.com/astral-sh/ruff/issues/8106.

**Comment by charliermarsh** [comment]
Merged!
